src/OrleansRuntime/Development/InMemoryLeaseProvider.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Orleans.LeaseProviders;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime.Development
+{
+    public class InMemoryLeaseProvider : ILeaseProvider
+    {
+        private readonly IDevelopmentLeaseProviderGrain leaseProvider;
+
+        public InMemoryLeaseProvider(IGrainFactory grainFactory)
+        {
+            this.leaseProvider = grainFactory.GetGrain<IDevelopmentLeaseProviderGrain>(0);
+        }
+
+        public async Task<AcquireLeaseResult[]> Acquire(string category, LeaseRequest[] leaseRequests)
+        {
+            try
+            {
+                return await this.leaseProvider.Acquire(category, leaseRequests);
+            } catch(Exception ex)
+            {
+                return leaseRequests.Select(request => new AcquireLeaseResult(null, ResponseCode.TransientFailure, ex)).ToArray();
+            }
+        }
+
+        public Task Release(string category, AcquiredLease[] acquiredLeases)
+        {
+            return this.leaseProvider.Release(category, acquiredLeases);
+        }
+
+        public async Task<AcquireLeaseResult[]> Renew(string category, AcquiredLease[] acquiredLeases)
+        {
+            try
+            {
+                return await this.leaseProvider.Renew(category, acquiredLeases);
+            }
+            catch (Exception ex)
+            {
+                return acquiredLeases.Select(request => new AcquireLeaseResult(null, ResponseCode.TransientFailure, ex)).ToArray();
+            }
+        }
+    }
+
+    public interface IDevelopmentLeaseProviderGrain : ILeaseProvider, IGrainWithIntegerKey
+    {
+    }
+
+    public class DevelopmentLeaseProviderGrain : Grain, IDevelopmentLeaseProviderGrain
+    {
+        private readonly Dictionary<Tuple<string, string>, Lease> leases = new Dictionary<Tuple<string, string>, Lease>();
```

Comment:
I don't necessarily disagree, but this would make this much more of a real implementation than I was shooting for.  If we actually store this state in storage, one can use this with a storage provider and use this in production, which is specifically **not** what this is for.

For test purposes, this is in-memory state (like the in-memory storage provider) and is not intended to be recoverable, so it should not be used with any recoverability tests.

Let's think on this..

Reply:
offline discussed deciding no need to add state now. since this should be enough for testing